### PR TITLE
bcn PP work Brad and Alex Buchanan all except bonus..for now

### PR DESCRIPTION
**Pull request by bradford-le**
bcn day 2 pp work Brad Le and Alex Buchanan

### CODE REVIEW

**Comment by bradford-le** [comment]
Brad Le and Alex Buchanan

File: starter-code/functions-and-arrays.js [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
+  var wordstore= words[0];
+  for(i=0; i<words.length;i+=1) {
+  if (wordstore.length < words[i].length) {
+    wordstore = words[i];
+    console.log("wordstore for greater loop: " + wordstore );
+  } else {
+    console.log("wordstore for else loop: " + wordstore );
+  }}
+return wordstore;
 }
```
Comment (tawebbcn) [inline starter-code/functions-and-arrays.js]: your solution is ok but you could try this with a forEach loop:

function findLongestWord (words) {
  var longest = "";

  words.forEach (function (word) {
    if (word.length > longest.length) {
      longest = word;
    }
  });

  return longest;
}

File: starter-code/functions-and-arrays.js [hunk 2]
```diff
@@ -23,27 +33,40 @@ var words = [
...
 console.log(total);
 
 // Calculate the Average
+var sumTotal = 0;
 function averageNumbers (array) {
-
+for(i=0; i<array.length;i+=1){
+   sumTotal += array[i];
+ }
+ return sumTotal/=array.length;
```
Comment (tawebbcn) [inline starter-code/functions-and-arrays.js]: You could call the function you did before to calculate the sum and then divide it for the array.length, you can also use the reduce method for calculating the sum.
Your solution is good though :)

File: starter-code/functions-and-arrays.js [hunk 3]
```diff
@@ -129,9 +175,15 @@ console.log(howManyMatter);
...
 // Bonus Quest
 function greatestProduct (matrix) {
-
+  var maxproduct;
+  var centervalue;
+  for(i=0;i<matrix.length;i+=1){
+    centervalue = Math.max.apply(Math, matrix);
+  }
+  return maxproduct;
 }
```
Comment (tawebbcn) [inline starter-code/functions-and-arrays.js]: Good job! Try to go for the bonus! ;)

**Pull request closed without merging**